packages/react-popup/Spec.md

```diff
@@ -34,30 +379,104 @@ _List the **Props** and **Slots** proposed for the component. Ideally this would
+
+#### Context menu
+
+Right click on the trigger opens the popup. The popup anchors to the mouse position.
+
+Shift + F10 on the trigger opens the popup
+
+#### Focus
+
+> TODO Discuss: This is a v0 behaviour not sure if it is still applicable
+
+When the trigger is focused, the popup opens.
+
+### Additional interactions
+
+#### Escape key
+
+The popup closes with the escape key when the trigger or popup content has focus.
+
+#### Click outside
+
+The popup closes when a click happens outside the popup trigger or content.
+
+### Focus trap
+
+When the popup is configured to be a focus trap, focus the first focusable element inside the popup on open.
+
+### Nesting
+
+Popups should be nested
 
 ## Accessibility
 
-Base accessibility information is included in the design document. After the spec is filled and review, outcomes from it need to be communicated to design and incorporated in the design document.
-
-- Decide whether to use **native element** or folow **ARIA** and provide reasons
-- Identify the **[ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/) pattern** and, if the component is listed there, follow its specification as possible.
-- Identify accessibility **variants**, the `role` ([ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions)) of the component, its `slots` and `aria-*` props.
-- Describe the **keyboard navigation**: Tab Oder and Arrow Key Navigation. Describe any other keyboard **shortcuts** used
-- Specify texts for **state change announcements** - [ARIA live regions
-  ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (number of available items in dropdown, error messages, confirmations, ...)
-- Identify UI parts that appear on **hover or focus** and specify keyboard and screen reader interaction with them
-- List cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)
-- List cases when **focus** needs to be **moved programatically** (if parts of the UI are appearing/disappearing or other cases)
+### Existing patterns
+
+The [WAI Dialog pattern](https://www.w3.org/TR/wai-aria-practices-1.2/#dialog_modal) and its variants are the inspirations for `Popup` accessibility.
+
+- [Datepicker](https://www.w3.org/TR/wai-aria-practices-1.2/examples/dialog-modal/datepicker-dialog.html)
+- [Modal Dialog](https://www.w3.org/TR/wai-aria-practices-1.2/examples/dialog-modal/dialog.html)
+
+### DOM element usage
+
+Only the `PopupContent` component will render DOM markup. By default the components renders an HTML `div` element.
+
+### Accessible markup
+
+Accessible markup is divided into two scenarios:
+
+```tsx
+// Popup that does not trap focus
+<button aria-haspopup="dialog">Trigger</button>
+<div role="complementary">
+  No focus trap
+</div>
+
+// Popup that does trap focus
+<button aria-haspopup="dialog">Trigger</button>
+<div role="dialog" aria-modal="true">
+  Focus trapped
+</div>
+```
```

Comment:
https://github.com/microsoft/fluentui/blob/a48d8dfd969f944c44b8c902cc9d418c333d063d/packages/fluentui/react-northstar/src/components/Popup/Popup.tsx#L528-L531

As I remember, `trapFocus` and `inline` causes accessibility issues and N* even has a warning about it. I think that we should follow up on this and disable this scenario if it causes accessibility issues.

Reply:
Yeah, currently working on `Tabster` for a lower level solution to the problem, that could even enable `trapFocus` and `inline` together. I mentioned the a11y requirement of `aria-hidden` in the spec.

At least in N* it would be an issue because `aria-hidden` is set by the `FocuzTrapZone` which only applies to body direct children. In tabster, it's an open question whether to do the same, or actually support the inline focus trap by walking up the tree and setting `aria-hidden` on siblings